[Chore] Arrow recipe build failed in MacOS with homebrew installed protoc (v27+)

### Chore Type

Dependency Update (Conan/Third-party)

### Task Description

`make release_spark_with_test` fails when a newer system-installed protoc (v27+) is present (e.g., via Homebrew). Arrow's substrait module compiles .proto files using the system protoc instead of the Conan-provided protoc (v3.21.4). The newer protoc generates code that includes
  `google/protobuf/runtime_version.h, a header that doesn't exist in protobuf 3.x`, causing:
```
[  9%] Building CXX object src/arrow/CMakeFiles/arrow_objlib.dir/array/data.cc.o
[  9%] Building CXX object CMakeFiles/substrait.dir/substrait_ep-generated/substrait/algebra.pb.cc.o
In file included from /Users/zhangxffff/.conan2/p/b/arrowe311c24581fbd/b/build/Release/substrait_ep-generated/substrait/algebra.pb.cc:6:
/Users/zhangxffff/.conan2/p/b/arrowe311c24581fbd/b/build/Release/substrait_ep-generated/substrait/algebra.pb.h:15:10: fatal error: 'google/protobuf/runtime_version.h' file not found
   15 | #include "google/protobuf/runtime_version.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  9%] Building CXX object src/arrow/CMakeFiles/arrow_objlib.dir/array/diff.cc.o
[  9%] Building CXX object src/arrow/CMakeFiles/arrow_objlib.dir/array/util.cc.o
1 error generated.
make[4]: *** [CMakeFiles/substrait.dir/substrait_ep-generated/substrait/algebra.pb.cc.o] Error 1
make[3]: *** [CMakeFiles/substrait.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....

```

### Motivation / Benefit

fix arrow build failed in MacOS with protoc installed with homebrew

### Related Files (Optional)

_No response_

### Implementation Plan (Optional)

_No response_

### Additional Context

_No response_

--- a/scripts/conan/patches/arrow.15.0.1-csv-support.patch
+++ b/scripts/conan/patches/arrow.15.0.1-csv-support.patch
@@ -449,7 +449,7 @@ index 24a7af89b..27655cdf4 100644
  
          if self.options.get_safe("skyhook", False):
              raise ConanInvalidConfiguration("CCI has no librados recipe (yet)")
-@@ -248,16 +388,11 @@ class ArrowConan(ConanFile):
+@@ -248,16 +388,13 @@ class ArrowConan(ConanFile):
              if self.dependencies["jemalloc"].options.enable_cxx:
                  raise ConanInvalidConfiguration("jemmalloc.enable_cxx of a static jemalloc must be disabled")
  
@@ -467,6 +467,8 @@ index 24a7af89b..27655cdf4 100644
 -        else:
 +        if Version(self.version) >= "13.0.0":
              self.tool_requires("cmake/[>=3.16 <4]")
++        if self.options.with_protobuf:
++            self.tool_requires("protobuf/<host_version>")
  
      def source(self):
 @@ -292,7 +427,7 @@ class ArrowConan(ConanFile):

